packages/langium/src/test/langium-test.ts

```diff
@@ -30,7 +30,27 @@ export function parseHelper<T extends AstNode = AstNode>(services: LangiumServic
     };
 }
 
-export type ExpectFunction = (actual: unknown, expected: unknown) => void;
+export type ExpectFunction = (actual: unknown, expected: unknown, message?: string) => void;
+
+let expectedFunction: ExpectFunction = (actual, expected, message) => {
```

Comment:
I have mixed feelings about writing an abstraction around expect like this. It only handles the `toEqual` case, meaning if we want to use other matchers like `toBe` or `toBeTruthy` we have to change the approach.

If the issue is related  to the `expect` function, we could try something like this instead to verify the function is present, apply a message when we can, and still allow adding a matcher of our choice.
```typescript
function expectedFunction(actual: unknown, message?: string) {
    if (typeof expect === 'function') {
        if (message && expect.length === 2) {
            // With `jest-expect-message`
            return expect(actual, message);
        } else {
            return expect(actual);
        }
    } else {
        throw new Error('No expect function provided');
    }
}
```

Reply:
Since we only use this function internally in the `langium-test` file, we usually only require the `toEqual` case. It restricts some special assertions (such as having to invert `not` assertions), but it makes testing in general both for us and the user easier.